### PR TITLE
Make fuzzy find popup scrollable

### LINKED ISSUE #1732: make fuzzy-find-popup scrollable
https://github.com/extrawurst/gitui/assets/55311933/52e561d3-2a08-4813-bb91-c5657e161a82

I think making fuzzy-find-popup scrollable will be natural.

Or, the selection should not be out of frame when moving up and down.

### PR DESCRIPTION
**Pull request by AmmarAbouZor**
This Pull Request fixes/closes #1732.

It changes the following:
- It makes the fuzzy-find popup scrollable

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

File: src/components/fuzzy_find_popup.rs [hunk 1]
```diff
@@ -232,13 +232,24 @@ impl DrawableComponent for FuzzyFindPopup {
 			self.find_text.draw(f, chunks[0])?;
 
 			if any_hits {
+				// Block has two lines up and down which need to be considered
+				const HEIGHT_BLOCK_MARGINE: usize = 2;
```
Comment (extrawurst) [inline src/components/fuzzy_find_popup.rs:236]: typo: *margin*

Comment (AmmarAbouZor) [reply src/components/fuzzy_find_popup.rs]: The borrow checker feels much simpler when we compare it with the English silent e rules :smiling_face_with_tear:

**Comment by extrawurst** [comment]
can you add also a scrollbar according to other scrollable content?

**Comment by AmmarAbouZor** [comment]
Scrollbar has been added and it will be rendered only when needed (When the matches list length is bigger than the list height)

**Comment by extrawurst** [comment]
Screenshot 2023-06-25 at 18 44 48

* [ ] something seems off if you compare the two colors
* [ ] the scrollbar should go on the border as in other examples

**Comment by AmmarAbouZor** [comment]
Merge conflicts with master has been resolved :+1:

**Comment by extrawurst** [comment]
Thank you ❤️

**Pull request merged**